test/fixtures/plugin-mongodb-core.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+var traceUtil = require('../../src/util.js');
+var shimmer = require('shimmer');
+var semver = require('semver');
+
+var api;
+
+var SUPPORTED_VERSIONS = '1 - 2';
+
+/**
+ * Wraps the provided callback so that the provided span will
+ * be closed immediately after the callback is invoked.
+ *
+ * @param {Span} span The span to be closed.
+ * @param {Function} done The callback to be wrapped.
+ * @return {Function} The wrapped function.
+ */
+function wrapCallback(transaction, done) {
+  return function(err, res) {
+    if (api.config.enhancedDatabaseReporting) {
+      if (err) {
+        // Errors may contain sensitive query parameters.
+        transaction.addLabel('mongoError', err);
+      }
+      if (res) {
+        var result = res.result ? res.result : res;
+        transaction.addLabel('results', traceUtil.stringifyPrefix(
```

Comment:
We should figure out whether/how to expose this through the API. We recently added truncation of label values to `addLabel`. I don't think we need two mechanisms for this stacked on top of each other. Maybe we can drop `databaseResultReportingSize` and summarize using `stringPrefix` and the server label value limit instead of truncating as done today. This would also simplify the plugin API.